cmd/state/stateless/state.go

```diff
@@ -226,44 +251,35 @@ beginTx:
 			return err
 		}
 
-		skipFirst := len(r.HistoryKey) > 0 // snapshot stores last analyzed key
 		for k, vIsEmpty, err := c.SeekKey(r.HistoryKey); k != nil || err != nil; k, vIsEmpty, err = c.NextKey() {
 			if err != nil {
 				return err
 			}
-			if skipFirst {
```

Comment:
Won't things be double-counted now you removed the `skipFirst` ?

Reply:
I moved tx.commit() from bottom of loot to top (it’s inside r.interrupt() ). If it on top, then we saving key which not processed yet. Then no reason to skip it when start from save-point.